compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -5024,6 +5034,9 @@ public boolean hasFillerValue(BType type) {
                 return checkFillerValue((BRecordType) type);
             case TypeTags.TUPLE:
                 BTupleType tupleType = (BTupleType) type;
+                if (tupleType.isCyclic) {
```

Comment:
We can still have `[]` as a valid constructor for cyclic tuples. Therefore we cannot assume that all cyclic tuples does not have a filler value.
http://localhost/~rukshanp/ballerina-spec/lang/build/spec.html#FillMember